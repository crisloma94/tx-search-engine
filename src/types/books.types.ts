// Defined in DEFAULT_BOOK_FIELDS
export interface BookSearchModel {
  author_name?: string[];
  cover_i?: number;
  first_publish_year?: number;
  id_goodreads?: string[];
  id_librarything?: string[];
  isbn?: string[];
  key: string;
  oclc?: string[];
  publisher?: string[];
  title: string;
}
