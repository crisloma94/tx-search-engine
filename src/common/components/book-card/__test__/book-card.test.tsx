import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import BookCard from '../book-card.component';
import { BookSearchModel } from '../../../../types/books.types';
import Theme from '../../../../styles/theme';
import { store } from '../../../../core/store/store';

describe('Book card component test suite', () => {
  const defaultBook: BookSearchModel = {
    author_name: ['Author name test'],
    cover_i: 5678,
    first_publish_year: 1984,
    id_goodreads: ['1234'],
    id_librarything: ['567'],
    isbn: ['89'],
    key: '/works/OL545572W',
    oclc: ['1', '2'],
    publisher: ['Publisher example'],
    title: 'Title test',
  };
  const getComponent = () => (
    <Provider store={store}>
      <Theme>
        <BookCard book={defaultBook} />
      </Theme>
    </Provider>
  );
  const setup = () => render(getComponent());

  it('Should match snapshot', () => {
    const { container } = setup();
    expect(container).toMatchSnapshot();
  });
});
