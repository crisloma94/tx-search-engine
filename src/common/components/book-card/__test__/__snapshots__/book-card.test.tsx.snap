// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Book card component test suite Should match snapshot 1`] = `
<div>
  <li
    class="sc-gTRrQi gGrCkx"
  >
    <img
      alt="Default book cover"
      class="sc-bgqQcB ccLtkH book-cover"
      src="https://covers.openlibrary.org/b/id/5678-M.jpg"
    />
    <div
      class="book-card__info"
    >
      Title test (1984)
      <p
        class="book-card__author"
      >
        Author name test
      </p>
      <p
        class="book-card__publisher"
      >
        Publisher example
      </p>
      <a
        href="https://www.amazon.es/s?k=Title test+Author name test"
        target="_blank"
      >
        Search on amazon
      </a>
    </div>
  </li>
</div>
`;
